mobilecoind/src/service.rs

```diff
@@ -4184,6 +4306,330 @@ mod test {
+        // those in resp
+        response_sci.tx_in.proofs = vec![];
+        assert_eq!(response_sci, generated_sci);
+
+        let tx = Tx::try_from(generate_mixed_tx_response.get_tx_proposal().get_tx()).unwrap();
+
+        assert_eq!(tx.prefix.outputs.len(), 4);
+
+        let mut found_counterparty_output = false;
+        let mut found_counterparty_change = false;
+        let mut found_originator_output = false;
+        let mut found_originator_change = false;
+        for output in tx.prefix.outputs.iter() {
+            if let Ok((amount, _scalar)) =
+                output.view_key_match(swap_counterparty.view_private_key())
+            {
+                if amount.token_id == Mob::ID {
+                    assert_eq!(amount.value, offered_value / 4);
+                    found_counterparty_output = true;
+                } else {
+                    assert_eq!(*amount.token_id, 1);
+                    // The quote asks for 123 eusd, counterparty fulfills to 1/4,
+                    // rounding up so they give away 31.
+                    // Change is their eusd utxo value minus that, minus the eusd transaction fee.
+                    let default_fee_val = get_test_fee_map()
+                        .get_fee_for_token(&TokenId::from(1))
+                        .unwrap();
+                    assert_eq!(
+                        amount.value,
+                        counterparty_eusd_utxo_value - 31 - default_fee_val
+                    );
+                    assert_eq!(tx.prefix.fee, default_fee_val);
+                    assert_eq!(tx.prefix.fee_token_id, 1);
+                    found_counterparty_change = true;
+                }
+            } else if let Ok((amount, _scalar)) =
+                output.view_key_match(swap_originator.view_private_key())
+            {
+                if amount.token_id == Mob::ID {
+                    assert_eq!(amount.value, offered_value * 3 / 4);
+                    found_originator_change = true;
+                } else {
+                    // 31 is 1/4 of 123, rounded up
+                    assert_eq!(amount, Amount::new(31, TokenId::from(1)));
+                    found_originator_output = true;
+                }
+            }
+        }
+        assert!(found_counterparty_output);
+        assert!(found_counterparty_change);
+        assert!(found_originator_output);
+        assert!(found_originator_change);
+
+        // Changing the fee in the request should work, and slightly adjust the output
+        // values.
+        let fee_override = 500_000;
+        request.set_fee(fee_override);
+
+        let generate_mixed_tx_response = client.generate_mixed_tx(&request).unwrap();
+        let mut response_sci = SignedContingentInput::try_from(
+            generate_mixed_tx_response.get_tx_proposal().get_scis()[0].get_sci(),
+        )
+        .unwrap();
+        // The request sci doesn't have proofs, so to test equal, we have to delete
+        // those in resp
```

Comment:
maybe this is the wrong approach, maybe the response shouldn't contain these proofs either. they are already elsewhere in the Tx anyways, the server is duplicating them I guess